module/applications/actor/base-sheet.mjs

```diff
@@ -686,18 +684,27 @@ export default class ActorSheet5e extends ActorSheet {
   /* -------------------------------------------- */
 
   /**
-   * Handle input changes to numeric form fields, allowing them to accept delta-typed inputs
+   * Handle input changes to numeric form fields, allowing them to accept delta-typed inputs, uses REGEX to determine if input is valid or not.
    * @param {Event} event  Triggering event.
    * @private
    */
+  // Uses strict regex validation to prevent malicious injection
+  // Has a limit on size to further prevent malicious injection, it's going to be hard to do malicious code in  10 characters.
   _onChangeInputDelta(event) {
     const input = event.target;
     const value = input.value;
-    if ( ["+", "-"].includes(value[0]) ) {
-      let delta = parseFloat(value);
-      input.value = foundry.utils.getProperty(this.actor, input.name) + delta;
+    if(value.match(/^([\d+-]\d*$)/)){
+      if ( ["+", "-"].includes(value[0]) ) {
+        let delta = parseFloat(value);
+        input.value = Number(foundry.utils.getProperty(this.actor, input.name)) + delta;
+      } else if ( value[0] === "=" ) input.value = value.slice(1);
+    }else if(value.match(/^([\d+-]\d*[+-/*]?[-]?\d*$)/)){
+      if(value.length <= 10){ 
+        input.value = Math.round(eval(value));
+      }
+    }else{
```

Comment:
It looks like this is checking for something like a formula and evaluating it? What is the use-case for this? Also, your PR is targeted at fixing a bug, and this seems to be additional behaviour that was not previously in the system. If you want to add functionality, please raise an issue separately with the intended use-case and submit a separate PR for it.

Reply:
Good idea

Reply:
Part of why this is part of the same thing, is that this requires the logic I built in the original PR.

Reply:
I have removed that section and placed it in new PR, #1897 .